crates/core/tedge/src/cli/mqtt/subscribe.rs

```diff
@@ -68,11 +104,11 @@ fn subscribe(cmd: &MqttSubscribeCommand) -> Result<(), MqttError> {
                 eprintln!("INFO: Connected");
                 client.subscribe(cmd.topic.as_str(), cmd.qos).unwrap();
             }
+            // TODO: should we keep trying to reconnect for all errors, or just
+            // if the broker isn't up and abort when e.g. we receive connection
+            // refused?
             Err(err) => {
                 let err_msg = err.to_string();
-                if err_msg.contains("I/O: Connection refused (os error 111)") {
-                    return Err(MqttError::ServerConnection(err_msg));
-                }
```

Comment:
I'm okay with the current proposal (to keep trying reconnect whatever the connection error) as the aim of  `tedge mqtt` is not to replace a full-fledged MQTT command line tool but just to provide a convenient way to interact with thin-edge leveraging the same configuration.